tendermint/src/lite/verifier.rs

```diff
@@ -66,38 +100,40 @@ where
     }
 
     // ensure that +2/3 validators signed correctly
-    verify_commit_full(&validators, commit)
+    verify_commit_full(&validators, &commit)
 }
 
 /// Verify the commit is trusted according to the last validators and is valid
 /// from the current validators for the header.
-pub fn verify_trusting<H, V, C>(
+pub fn verify_trusting<H, V, C, L>(
     header: H,
     commit: C,
     last_validators: V,
     validators: V,
+    trust_level: L,
 ) -> Result<(), Error>
 where
     H: Header,
     V: ValidatorSetLookup,
     C: Commit,
+    L: TrustLevel,
 {
     // NOTE it might be more prudent to do the cheap validations first
     // before we even call verify_commit_trusting, but not doing that
     // makes the code cleaner and allows us to just call verify directly.
 
     // ensure that +1/3 of last trusted validators signed correctly
-    if let Err(e) = verify_commit_trusting(&last_validators, &commit) {
+    if let Err(e) = verify_commit_trusting(&last_validators, &commit, trust_level) {
```

Comment:
Is this function supposed to work for adjacent headers as well? For those we need a different check, as it is the case in `verify()`. It is not clear to me why we need both `verify()` and `verify_trusting()`.
Can't we use only `verify()` which in the case of sequential is always called with sequential (non skipping) headers while in the case of bisection with both skipping and sequential headers?

Reply:
Thanks @ancazamfir. I think we can delete `verify_trusting`. I shamelessly adapted some naming from the current go-code without completing it yet (and `verify_trusting` is remainder of that incomplete refactoring). Sorry for the confusion.

I think in this PR it make sense to adapt the naming and logic of the spec instead. In particularly what @milosevic  has been working on here: https://github.com/tendermint/spec/blob/4f7c55507cb99c35c7774c583f182859283f3017/spec/consensus/light-client.md

Reply:
Deleted `verify_trusting` for now in 3d49482